### PR TITLE
dev-util/ctags: add a snapshot ebuild

### PR DESCRIPTION
**Pull request by marcecj**
Add a snapshot ebuild based on the "sourceforge" branch of universal-ctags,
which is based on the original exuberant-ctags SVN repository.  This fixes
several upstream bugs, including crashes such as
http://sourceforge.net/p/ctags/bugs/359/, and obsoletes one Gentoo patch.

I have two questions:
1. Is eautoreconf correct here? I'm not particularly knowledgeable about autotools.
2. Is it OK to use the universal-ctags "sourceforge" branch? I thought it would be a bit more forward thinking to use that rather than rely on the SourceForge hosted SVN repo, or is it a tad paranoid, or does it just not matter ;-) ?

This obsoletes PR #208.

### CODE REVIEW

**Comment by mgorny** [comment]
@radhermit

**Comment by Coacher** [comment]
Just passing by...

> 1. Is eautoreconf correct here?

Yes, because there is no configure script, nor Makefile. Only configure.ac and Makefile.in files.

Also you could probably inherit autotools eclass first and then eliminate src_unpack(). I wonder where you use autotools eclass functionality at all though.

**Comment by marcecj** [comment]
On Friday 30 October 2015 09:16:35 Coacher wrote:
>Just passing by...
>
>> 1. Is eautoreconf correct here?
>
>Yes, because there is no configure script, nor Makefile. Only configure.ac
>and Makefile.in files.

OK, thanks.

>Also you could probably inherit autotools eclass first and then eliminate
>src_unpack(). I wonder where you use autotools eclass functionality at all
>though.

Hmm, it may make `git-r3_src_unpack` unnecessary, but then it seems as if the 
default `src_unpack` is never called, which is necessary for `USE="ada"`.  And 
doesn't the autotools eclass define `eautoreconf`?

**Comment by Coacher** [comment]
> Hmm, it may make `git-r3_src_unpack` unnecessary, but then it seems as if the default `src_unpack` is never called, which is necessary for `USE="ada"`. And doesn't the autotools eclass define `eautoreconf`?

Yes, you are right. I missed this.

**Comment by mgorny** [comment]
Oh, sorry for replying this late. When we talked about snapshot ebuild, we meant fetching snapshot via http(s), not git. I mean the github 'archive' URL + vcs-snapshot eclass to unpack it.

**Comment by marcecj** [comment]
Ah! I was unaware of that GitHub feature.

**Comment by marcecj** [comment]
OK, the ebuild now uses the vcs-snapshot eclass.

File: dev-util/ctags/ctags-20140919.ebuild [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+src_install() {
+	emake prefix="${D}"/usr mandir="${D}"/usr/share/man install
+
+	# namepace collision with X/Emacs-provided /usr/bin/ctags -- we
+	# rename ctags to exuberant-ctags (Mandrake does this also).
+	mv "${D}"/usr/bin/{ctags,exuberant-ctags} || die
+	mv "${D}"/usr/share/man/man1/{ctags,exuberant-ctags}.1 || die
+
+	dodoc FAQ NEWS README
+	dohtml EXTENDING.html
```
Comment (mgorny) [inline dev-util/ctags/ctags-20140919.ebuild]: `dohtml` is mostly deprecated, please use `dodoc` here.

**Comment by mgorny** [comment]
Also please don't use `-YYYYMMDD` versioning as that's going to cause a real pain if we do another release in the future. Instead, use `X.Y_pYYYYMMDD` or `X.Y_preYYYYMMDD` whichever is more appropriate, taking X.Y from nearby release.

File: dev-util/ctags/ctags-20140919.ebuild [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="5"
+
+inherit eutils autotools vcs-snapshot
+
+DESCRIPTION="Exuberant Ctags creates tags files for code browsing in editors"
+HOMEPAGE="http://ctags.sourceforge.net"
+SRC_URI="https://github.com/universal-ctags/ctags/archive/9fce9dd0afd3dd261c681825a61d3e9ffcaa7eea.tar.gz -> ${P}.tar.gz
```
Comment (mgorny) [inline dev-util/ctags/ctags-20140919.ebuild]: would be nice to add a comment about the branch here, so people know it's not from master.

**Comment by marcecj** [comment]
I updated the ebuild following @mgorny's comments.  The hashes of the tarball changed because for some reason the directory in the old tarball was named `ctags-sourceforge` instead of `ctags-<hash>`.  I wonder why, because the ebuild wouldn't have worked otherwise due to the way the vcs-snapshot eclass works (and I always test before I push).

**Comment by mgorny** [comment]
Thanks. Just FYI, I've discussed this more and we'd probably be interested in replacing it with universal-ctags. However, it's not yet clear how to add the new package. Do you happen to know if there will be releases for universal-ctags and/or if they're planning to continue exuberant-ctags version numbers?

**Comment by marcecj** [comment]
You mean this snapshot ebuild (though I admit that I find it annoying that they didn't add release tags to the sourceforge branch)?  As to your question: I honestly have no idea.  In [bug \#561074](https://bugs.gentoo.org/show_bug.cgi?id=561074) @Coacher expressed the opinion (and I agree) that introducing universal-ctags is not a good idea right now.  AFAIK the best you can do at the moment is to keep track of how many [release blockers](https://github.com/universal-ctags/ctags/labels/Release%20blocker) remain.

**Pull request merged**